Traducir.Web.Net/Scripts/shared/utils.ts

```diff
@@ -0,0 +1,29 @@
+export function clone(object) {
+    return JSON.parse(JSON.stringify(object));
+}
+
+export function spinner(show) {
+    document.getElementById("spinner").style.display = show ? "block" : "none";
+}
+
+export function dynamicEventHook(events, selector, handler) {
+    if (!Array.isArray(events)) {
+        events = [events];
+    }
+
+    for (const event of events) {
+        document.addEventListener(event, e => {
```

Comment:
@Konamiman I was puzzled when debugging this... from what I see, we're setting all the events on the document and then when the event is triggered, we check if the target is in the selector.

I guess the reasoning here is to avoid having to hook elements dynamically created with the events... but I'm a bit worried about this introducing unexpected effects... also, every time there's a click on the document (even if it's not in an element) we check each selector.

I can't help but feel we'd be better of with using events attached to the actual dom elements (it would also make it easier to debug it).

I'm not saying we need to tweak this right now, just that it's something I'd like to address (even post-merge).

What do you think?

Reply:
Yes, we check each selector on each click, but since this is a very small project I don't think there's a huge performance penalty or noticeable side effect, so I prioritized code simplicity. If/when we discover that this approach has indeed problems (or if the project grows), we can think about changing it.